base/libgit2/index.jl

```diff
@@ -46,43 +46,43 @@ end
 read_tree!(idx::GitIndex, hash::AbstractGitHash) =
     read_tree!(idx, GitTree(repository(idx), hash))
 
-function add!{T<:AbstractString}(idx::GitIndex, files::T...;
-             flags::Cuint = Consts.INDEX_ADD_DEFAULT)
+function add!(idx::GitIndex, files::AbstractString...;
```

Comment:
I think this should still work, but this change is slightly more permissive right? In the old version wouldn't each of `files` have to be the same type, but now they can individually be different `AbstractString` subtypes?